gyp/pylib/gyp/input.py

```diff
@@ -1222,6 +1222,8 @@ def ProcessVariablesAndConditionsInDict(the_dict, phase, variables_in,
     # Skip "variables", which was already processed if present.
     if key != 'variables' and type(value) is str:
       expanded = ExpandVariables(value, phase, variables, build_file)
+      if type(expanded) in (long, list, dict):
+        continue
```

Comment:
This doesn't look self-evidently correct to me.  What is yarn doing that npm doesn't that causes ExpandVariables to return unexpected types?  `long` is acceptable because `int` is accepted but why `list` and `dict`?  With what keys do you get those?